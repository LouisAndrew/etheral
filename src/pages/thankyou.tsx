import React from 'react';
import { PageProps } from 'gatsby';

import Thankyou, { Props } from 'templates/thankyou';
import { Layout } from 'components/layout';

/**
 * Page component for thankyou page
 * @param props State and Page query
 */
const ThankyouPage = (props: PageProps) => {
    const { state } = props.location;

    if (state) {
        return (
            <Layout>
                <Thankyou {...(state as Props)} />
            </Layout>
        );
    } else {
        return <></>;
    }

    // handle messages if: va, alfa, cstore usw.
};

export default ThankyouPage;
