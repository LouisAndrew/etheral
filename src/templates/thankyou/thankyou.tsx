import React from 'react';
// import styling libs
import { Box, Heading } from 'rebass';
// import local components

import thanks from './assets/desktop.jpg';
import thanksMobile from './assets/mobile.jpg';

type Props = {};

/**
 * Thankyou page. Shown after user completes a transaction
 * @param param0 Props
 */
const Thankyou: React.FC<Props> = () => {
    const THANKYOU_MSG =
        'THANK YOU FOR YOUR PURCHASE!\nCHECK YOUR EMAIL FOR YOUR ORDER CONFIRMATION';

    return (
        <Box className="content">
            {/* separated content container from the inner content */}
            <Box
                className="inner-wrapper"
                sx={{
                    position: 'relative',
                    overflow: 'hidden',
                    height: ['80vh', '120vh', '55vh', '65vh'],
                    minHeight: ['unset', 'unset', '400px'],
                    maxHeight: ['unset', '380px', '50vh', 'unset'],
                }}
                css={`
                    /* kinda specific for surface duo. */
                    @media screen and (min-width: 27em) and (max-width: 48em) and (orientation: portrait) {
                        max-height: 40vh !important;
                        margin-top: 16px !important;
                    }

                    /* kinda specific for ipad(s) */
                    @media screen and (min-width: 48em) and (max-width: 64em) and (orientation: portrait) {
                        max-height: 40vh !important;
                    }
                `}
            >
                <Box
                    className="bg"
                    sx={{
                        position: 'absolute',
                        top: 0,
                        left: 0,
                        height: '100%',
                        width: '100%',
                        zIndex: 1,
                        img: {
                            height: 'auto',
                            width: '100%',

                            '&.thx-mobile': {
                                display: ['block', 'none'],
                            },
                            '&.thx-desktop': {
                                display: ['none', 'block'],
                            },
                            transform: [
                                'unset',
                                'unset',
                                'unset',
                                'unset',
                                'translateY(-200px)',
                            ],
                        },
                    }}
                >
                    <img
                        alt="Thankyou"
                        src={thanksMobile}
                        className="thx-mobile"
                    />
                    <img alt="Thankyou" src={thanks} className="thx-desktop" />
                </Box>
                <Box
                    sx={{ zIndex: 2, position: 'relative' }}
                    py={[128]}
                    px={[4, 6, 7, 9]}
                >
                    <Heading
                        color="#fff"
                        fontWeight="semiBold"
                        textAlign={['center', 'left']}
                        fontSize={[1, 1, 5, 6, 7]}
                        my="auto"
                        mx={['auto', 0, 0]}
                    >
                        {THANKYOU_MSG}
                    </Heading>
                </Box>
            </Box>
        </Box>
    );
};

export { Thankyou };
