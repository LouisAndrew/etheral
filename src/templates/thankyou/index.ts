import { Thankyou, Props } from './thankyou';

export { Props };

export default Thankyou;
