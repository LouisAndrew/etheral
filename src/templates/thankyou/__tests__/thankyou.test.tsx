import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Thankyou } from '../thankyou';

describe('Thankyou', () => {
    const mockData = {
        total: 12000,
        paymentNo: 123456,
        paymentName: 'Louis',
        expired: '12-3-4',
        currency: 'IDR',
    };

    const Element = <Thankyou {...mockData} />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    it('should render all data(s) correctly correctly', () => {
        const { getByTestId } = render(Element);

        const Total = getByTestId('total');
        const PaymentNo = getByTestId('paymentNo');
        const PaymentName = getByTestId('paymentName');
        const Expired = getByTestId('expired');

        const { total, paymentNo, paymentName, expired } = mockData;

        expect(Total).toHaveTextContent(total.toString());
        expect(PaymentNo).toHaveTextContent(paymentNo.toString());
        expect(PaymentName).toHaveTextContent(paymentName);
        expect(Expired).toHaveTextContent(expired);
    });

    it('matches snapshot', () => {
        const run = false;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
