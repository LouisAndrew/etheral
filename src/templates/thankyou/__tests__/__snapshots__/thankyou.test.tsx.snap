// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thankyou matches snapshot 1`] = `
<div
  className="content css-vurnku"
>
  <div
    className="inner-wrapper css-iasgit"
  >
    <div
      className="bg css-neasi4"
    >
      <img
        alt="Thankyou"
        className="thx-mobile"
        src="test-file-stub"
      />
      <img
        alt="Thankyou"
        className="thx-desktop"
        src="test-file-stub"
      />
    </div>
    <div
      className="css-4xdyao"
    >
      <h2
        className="css-mk3a7k"
      >
        THANK YOU FOR YOUR PURCHASE!
CHECK YOUR EMAIL FOR YOUR ORDER CONFIRMATION
      </h2>
      <div
        className="css-r2xo4s"
      >
        Please transfer
         
        <strong>
          IDR
           
          12000
        </strong>
         
        before 
        <strong>
          12-3-4
        </strong>
         to:
        <br />
        Name: 
        <strong>
          Louis
        </strong>
        <br />
        No: 
        <strong>
          123456
        </strong>
      </div>
    </div>
  </div>
</div>
`;
